Implement torch.allclose for sharded tensor.

kumpera opened the pull request:
Summary:
Implement torch.allclose op for sharded tensors.

Test Plan:

Automated test added.

Reviewers:

pritamdamania87

Subscribers:

Tasks:

Tags:

Fixes #67112


cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

pritamdamania87 commented on torch/distributed/_sharded_tensor/ops/equals.py:
```diff
@@ -1,58 +1,7 @@
 import torch
```
We can get rid of this file and just put everything in `binary_cmp.py`?

kumpera replied:
done

pritamdamania87 commented on torch/distributed/_sharded_tensor/ops/allclose.py:
```diff
@@ -0,0 +1,7 @@
+import torch
```
Probably just have a single file for comparison ops like equal and allclose instead of a file per op.

kumpera replied:
Good point, will merge the two files.

pritamdamania87 commented on test/distributed/_sharded_tensor/ops/test_allclose.py:
```diff
@@ -0,0 +1,51 @@
+# Owner(s): ["oncall: distributed"]
```
Lets have a single test file for equals and allclose instead of a separate file for each.

pritamdamania87 replied:
Note that each test file needs to be added to run_test.py. For example: https://github.com/pytorch/pytorch/blob/master/test/run_test.py#L209 (look at all references of test_equals.py) there.

pritamdamania87 commented on torch/testing/_internal/distributed/_sharded_tensor/__init__.py:
```diff
@@ -75,6 +80,91 @@ def assert_sharded_tensor_equal(self, st1, st2):
         self.assertEqual(st1.sharding_spec(), st2.sharding_spec())
         self.assertEqual(len(st1.remote_shards()), len(st2.remote_shards()))
 
+class BinaryCmpOpTestBase(ShardedTensorTestBase):
```
nit: This can just be in `binary_cmp.py` since it is used only in one file?

kumpera replied:
good point.

kumpera commented:
The single failure looks spurious.

kumpera committed:
Implement torch.allclose for sharded tensor. (#70331)

Summary:
Implement torch.allclose op for sharded tensors.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/70331

Test Plan:
Automated test added.
pritamdamania87
Fixes https://github.com/pytorch/pytorch/issues/67112

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse SciPioneer H-Huang

Reviewed By: pritamdamania87

Differential Revision: D33339137

Pulled By: kumpera

fbshipit-source-id: 02dd02df7581903a7e7a34c57118550bc1535c79
